test/testdata/parser/error_recovery/unterminated_array.rb

```diff
@@ -0,0 +1,35 @@
+# typed: true
+
+class A
+  def f
+    [1,
+  #   ^ error: unexpected token ","
+  # ^ error: unterminated "["
+    puts "hi"
+  end
+  def g
+    [
+  # ^ error: unterminated "["
+  end
+  def h
+    puts "ho"
+  end
+end
+
+class B
+  def q
+    puts "ho"
+    [
+  # ^ error: unterminated "["
+  end
+  def r
+    puts "hi"
+  end
+  def s
+    puts ([] + [)
+             # ^ error: unterminated "["
+  end
+  def t
+    puts "hi"
+  end
+end
```

Comment:
Given the implementation above, I'm a little worried that this only does well on a handful of test cases. Some test cases I've found that don't look that great:

```ruby
# typed: true

class A
  X = [1,

  sig {void}
  def bar
  end
end
```

- drops the `1` from the array
- drops the whole sig + method def

```ruby
# typed: true

class A
  def bar
    puts 'before'
    x = [1,
    puts 'after'
  end
end
```

- again drops the `1` from the array
- drops the `puts 'after'` from the method body